email-common/src/main/java/io/commercetools/sunrise/email/MessageEditor.java

```diff
@@ -0,0 +1,21 @@
+package io.commercetools.sunrise.email;
+
+import javax.annotation.Nonnull;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+/**
+ * A functional interface for modifying a message without needing to handle checked exceptions like the
+ * {@link MessagingException}s thrown by {@link MimeMessage}.
+ */
+@FunctionalInterface
+public interface MessageEditor {
```

Comment:
This one makes sense, so we should keep it. But we still have the problem that if you don't read the documentation (which of course shouldn't be the case, but we know lazy developers ;)) you will lost without a clue of what to do with `MessageEditor`, which creates frustration and that's the worst first experience when you work with something new.
To minimize that, what if we provide a basic implementation of MessageEditor which expects the typical information for an email (i.e. from, recipients, subject, content). People looking for implementations of `MessageEditor` will find it and use it, once they realize they are missing something more advanced, they will think about creating their own implementation.
The pros are clear: easier getting started (easier to achieve basic usage), more intuitive, less fully relying on Java 8 knowledge (better for beginners)... cons are that we would be leaving as a background solution what is actually quite powerful itself and still simple (once you know it).
An alternative solution would be to document in this interface that you are supposed to implement it yourself and an example. People not knowing what to do will likely end up in this file looking for a clue. Actually I would do it nonetheless, also with the basic implementation solution.
@schleichardt @selo-ecube what do you think about it?